x-pack/test/detection_engine_api_integration/security_and_spaces/roles_users_utils/index.ts

```diff
@@ -64,6 +65,20 @@ export const createUserAndRole = async (
   }
 };
 
+/**
+ * Given a roleName and security service this will delete the roleName
+ * and user
+ * @param roleName The user and role to delete with the same name
+ * @param securityService The security service
+ */
+export const deleteUserAndRole = async (
+  securityService: ProvidedType<typeof services['security']>,
+  roleName: ROLES
+) => {
+  await securityService.role.delete(roleName);
+  await securityService.user.delete(roleName);
+};
```

Comment:
Nit: I would ask we switch the ordering of these delete calls such that we delete the user first and then the role. This would better reflect the setup of posting the role first and then the user.

Reply:
Yeah, it does seem wrong if I deleted a role first and the user had a dangling role. I am curious how it allowed me to do that in first place but you are correct this would be better for future proofing things and more correct.